### ISSUE TITLE
RFE: bwc-policy for Virtual Servers

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-appsvcs-extension#119]
##### Do you already have an issue opened with F5 support?
No

##### ISSUE TYPE

 - Feature Idea

##### AS3 BUILD/ VERSION

```
{
    "version": "3.13.0",
    "release": "3",
    "schemaCurrent": "3.13.0",
    "schemaMinimum": "3.0.0"
}
```

##### BIGIP VERSION
```
  Product     BIG-IP
  Version     13.1.1.2
  Build       0.0.4
  Edition     Point Release 2
  Date        Thu Oct 11 15:32:21 PDT 2018
```

##### SUMMARY

currently you can create a Bandwidth Control (BWC) Policy, but you are unable to attach it directly to a virtual server (it has to be referenced from a PEM policy).

##### STEPS TO REPRODUCE
Create a BWC policy, no mechanism to attach to virtual server.
##### EXPECTED RESULTS
Mechanism to attach to virtual server

example of tmsh command
```
modify virtual DemoApplication/serviceMain bwc-policy Shared/applicationName_BWCtlPolicy ...
```

##### ACTUAL RESULTS
No way to attach other than an iRule.  Here's an interim workaround example

```
{
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.0.0",
        "id": "urn:uuid:5cff22e8-d3d1-6056-8577-b0b193f812eb",
        "label": "Add_bandwidth_control_policy",
        "remark": "Add a bandwidth control policy",
        "DemoTenant": {
            "class": "Tenant",
            "Shared": {
                "class": "Application",
                "template": "shared",
                "enable": true,
                "applicationName_BWCtlPolicy": {
                    "class": "Bandwidth_Control_Policy",
                    "maxBandwidth": 1,
                    "maxBandwidthUnit": "Mbps"
                },
                "AttachBWC": {
                    "class": "iRule",
                    "iRule": "when CLIENT_ACCEPTED {\nBWC::policy attach /`T`/Shared/applicationName_BWCtlPolicy\n}"
                }
            },
            "DemoApplication": {
                "class": "Application",
                "template": "http",
                "serviceMain": {
                    "class": "Service_HTTP",
                    "remark": "Accepts HTTP/nonTLS connections on port 80 ",
                    "virtualAddresses": [
                        "10.1.10.10"
                    ],
                    "pool": "custom_pool",
                    "profileHTTP": {
                        "use": "custom_http_profile"
                    },
                    "iRules": [
                        "/@/Shared/AttachBWC"
                    ]
                },
                "custom_pool": {
                    "class": "Pool",
                    "monitors": [
                        "http"
                    ],
                    "members": [
                        {
                            "servicePort": 8080,
                            "serverAddresses": [
                                "192.168.128.11",
                                "192.168.128.12",
                                "192.168.128.13"
                            ]
                        }
                    ]
                },
                "custom_http_profile": {
                    "class": "HTTP_Profile",
                    "xForwardedFor": true
                }
            }
        }
    }
}
```

**Comment by focrensh** [comment]
AT-630 for internal tracking

**Comment by dstokesf5** [comment]
This was resolved in release 3.19.0.
